File: Python/Program/Shirt.py
from Clothing import Clothing

class Shirt(Clothing):
    def __init__(self, idProduct="", name="", brand="", price=0, size="", material="", gender="", color="", sleeveType=""):
        super().__init__(idProduct, name, brand, price, size, material, gender) #super untuk memanggil atribut di class product dan clothing
        self.color = color                                                      #atribut untuk menampung warna
        self.sleeveType = sleeveType                                            #atribut untuk menampung sleeve type

    #Getter dan Setter
    def setColor(self, color):
        self.color = color

    def setSleeveType(self, sleeveType):
        self.sleeveType = sleeveType

    def getColor(self):
        return self.color

    def getSleeveType(self):
        return self.sleeveType

File: PHP/Program/Shirt.php
<?php
require_once 'Clothing.php';

class Shirt extends Clothing {
    protected $color; // Warna kemeja
    protected $sleeve_type; // Jenis lengan kemeja

    // Konstruktor untuk inisialisasi objek kemeja
    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender, $color, $sleeve_type) {
        parent::__construct($idProduct, $name, $brand, $price, $size, $material, $gender);
        $this->color = $color;
        $this->sleeve_type = $sleeve_type;
    }

    // Getter untuk mendapatkan warna kemeja
    public function getColor() {
        return $this->color;
    }

    // Getter untuk mendapatkan jenis lengan kemeja
    public function getSleeveType() {
        return $this->sleeve_type;
    }

    // Setter untuk mengatur warna kemeja
    public function setColor($color) {
        $this->color = $color;
    }

    // Setter untuk mengatur jenis lengan kemeja
    public function setSleeveType($sleeve_type) {
        $this->sleeve_type = $sleeve_type;
    }

    // Method untuk mendapatkan informasi lengkap tentang kemeja
    public function getInfo() {
        return parent::getInfo() . ", Color: {$this->color}, Sleeve Type: {$this->sleeve_type}";
    }
}
?>
